lib/Twig/TokenStream.php

```diff
@@ -32,9 +32,19 @@ class Twig_TokenStream
      */
     public function __construct(array $tokens, $filename = null, $source = null)
     {
+        if (!$filename instanceof Twig_Source) {
+            if (null !== $filename || null !== $source) {
+                @trigger_error(sprintf('Passing a string as the $filename argument of %s() is deprecated since version 1.27. Pass a Twig_Source instance instead.', __METHOD__), E_USER_DEPRECATED);
+            }
+            $this->source = new Twig_Source($filename, $source);
```

Comment:
Please also add legacy tests covering the deprecated APIs. They would catch such mistakes (and forbid breaking BC by mistake)